File: samples/CslaNet/cs/ProjectTrackercs/PTWin/StatusBusy.cs
using System;
using System.Windows.Forms;

namespace PTWin
{
  public class StatusBusy : IDisposable
  {
    private string _oldStatus;
    private Cursor _oldCursor;

    public StatusBusy(string statusText)
    {
      _oldStatus = MainForm.Instance.StatusLabel.Text;
      MainForm.Instance.StatusLabel.Text = statusText;
      _oldCursor = MainForm.Instance.Cursor;
      MainForm.Instance.Cursor = Cursors.WaitCursor;
    }

    // IDisposable
    private bool _disposedValue = false; // To detect redundant calls

    protected void Dispose(bool disposing)
    {
      if (!_disposedValue)
        if (disposing)
        {
          MainForm.Instance.StatusLabel.Text = _oldStatus;
          MainForm.Instance.Cursor = _oldCursor;
        }
      _disposedValue = true;
    }

    public void Dispose()
    {
      // Do not change this code. Put cleanup code in Dispose(bool disposing) above
      Dispose(true);
      GC.SuppressFinalize(this);
    }
  }
}

File: samples/CslaNet/vb/ProjectTrackervb/PTWin/StatusBusy.vb
Public Class StatusBusy

  Implements IDisposable

  Private _oldStatus As String
  Private _oldCursor As Cursor

  Public Sub New(ByVal statusText As String)

    _oldStatus = MainForm.StatusLabel.Text
    MainForm.StatusLabel.Text = statusText
    _oldCursor = MainForm.Cursor
    MainForm.Cursor = Cursors.WaitCursor

  End Sub

  ' IDisposable
  Private disposedValue As Boolean = False ' To detect redundant calls

  Protected Overridable Sub Dispose(ByVal disposing As Boolean)
    If Not Me.disposedValue Then
      If disposing Then
        MainForm.StatusLabel.Text = _oldStatus
        MainForm.Cursor = _oldCursor
      End If
    End If
    Me.disposedValue = True
  End Sub

  Public Sub Dispose() Implements IDisposable.Dispose
    ' Do not change this code.  
    ' Put cleanup code in Dispose(ByVal disposing As Boolean) above.
    Dispose(True)
    GC.SuppressFinalize(Me)
  End Sub

End Class
